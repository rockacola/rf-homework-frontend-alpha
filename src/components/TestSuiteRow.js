import React, { useState } from "react";
import { Link } from "react-router-dom";

export const TestSuiteRow = ({ testSuite }) => {
  const [showDetails, setShowDetails] = useState(false);

  const onClick = () => {
    setShowDetails(!showDetails);
  };

  const chevron = showDetails ? "▼" : "▶";
  return (
    <>
      <tr
        data-test-suite-id={testSuite.id}
        style={{ cursor: "pointer" }}
        onClick={onClick}
      >
        <td>{chevron}</td>
        <td>{testSuite.test_suite_name}</td>
        <td style={{ textAlign: "right" }}>
          {testSuite.test_plans.length} tests
        </td>
        <td style={{ paddingLeft: "10px", paddingRight: "10px" }}>
          <Link to={`/test_suites/${testSuite.id}/edit`}>[Edit]</Link>
        </td>
      </tr>

      {showDetails && (
        <tr>
          <td colspan="4">
            <table border="0" cellspacing="0" cellpadding="0">
              <tbody>
                {testSuite.test_plans.map((tp) => {
                  return (
                    <tr>
                      <td style={{ width: "150px" }}></td>
                      <td style={{ width: "350px" }}>{tp.test_name}</td>
                      <td style={{ width: "150px" }}>{tp.browser}</td>
                      <td style={{ width: "50px" }}>{tp.instruction_count}</td>
                    </tr>
                  );
                })}
              </tbody>
            </table>
          </td>
        </tr>
      )}
    </>
  );
};
