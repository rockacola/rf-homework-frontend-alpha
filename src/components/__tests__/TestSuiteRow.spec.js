import React from "react";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import { TestSuiteRow } from "../TestSuiteRow";

describe("TestSuiteRow", () => {
  let comp, props;

  beforeEach(() => {
    props = {
      testSuite: {
        id: 19,
        test_suite_name: "Suite Coffee Brown Log",
        test_plans: [
          {
            test_name: "Test Plan Forget Beat Tent",
            browser: "firefox",
            instruction_count: 4,
          },
        ],
      },
    };
    comp = renderer.create(
      <MemoryRouter>
        <TestSuiteRow {...props} />
      </MemoryRouter>
    );
  });

  it("renders", () => {
    expect(comp).toMatchSnapshot();
  });
});
