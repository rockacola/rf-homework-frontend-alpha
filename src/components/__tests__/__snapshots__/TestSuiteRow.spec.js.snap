// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestSuiteRow renders 1`] = `
<tr
  data-test-suite-id={19}
  onClick={[Function]}
  style={
    Object {
      "cursor": "pointer",
    }
  }
>
  <td>
    â–¶
  </td>
  <td>
    Suite Coffee Brown Log
  </td>
  <td
    style={
      Object {
        "textAlign": "right",
      }
    }
  >
    1
     tests
  </td>
  <td
    style={
      Object {
        "paddingLeft": "10px",
        "paddingRight": "10px",
      }
    }
  >
    <a
      href="/test_suites/19/edit"
      onClick={[Function]}
    >
      [Edit]
    </a>
  </td>
</tr>
`;
